src/theme/index.ts

```diff
@@ -91,12 +72,11 @@ const extendTheme = ({
 }: {
   baseTheme?: typeof theme
   extendedTheme: RecursivePartial<typeof theme>
-}) => deepmerge(baseTheme, extendedTheme)
+}) => deepmerge(baseTheme, extendedTheme) as typeof theme
```

Comment:
deep merge does not accept a generic ?

Reply:
@chambo-e It does but both arguments doesn't have same types, first argument is of type `typeof theme` and second argument is `RecursivePartial<typeof theme>`.

If I put `typeof theme` as generic it will trow an error on `extendedTheme` parameter, that it's type `RecursivePartial<typeof theme>` and not `typeof theme`.

I could put `RecursivePartial<typeof theme>` as generic, but then `extendTheme` function will return type `RecursivePartial<typeof theme>` and that's not what we want, we want returned type `typeof theme` for our function `extendTheme`, as it is what the function return.

So that's the solution I found to be best working in this case

Reply:
Sorry for long post, here is a potato

![potato](https://media.giphy.com/media/U1rlk8zdcAwbm/giphy.gif)